src/test/java/org/broadinstitute/consent/http/resources/InstitutionResourceTest.java

```diff
@@ -0,0 +1,160 @@
+    mockInstitution.setCreateDate(new Date());
+    mockInstitution.setCreateUser(1);
+    mockInstitution.setUpdateDate(new Date());
+    mockInstitution.setUpdateUser(1);
+    mockInstitution.setId(1);
+    return mockInstitution;
+  }
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    resource = new InstitutionResource(userService, institutionService, institutionUtil);
+  }
+
+  private Institution initInsitutionModel() {
+    Institution mockInstitution = new Institution();
+    mockInstitution.setName("Test Name");
+    return mockInstitution;
+  }
+
+  private void initResource() {
+    resource = new InstitutionResource(userService, institutionService, institutionUtil);
+  }
+
+  @Test
+  public void testGetInstitutionsForAdmin() {
+    List<Institution> institutions = new ArrayList<Institution>();
+    Institution mockInstitution = institutionSetup();
+    institutions.add(mockInstitution);
+    when(userService.findUserByEmail(anyString())).thenReturn(adminUser);
+    when(institutionService.findAllInstitutions()).thenReturn(institutions);
+    initResource();
+    Response adminResponse = resource.getInstitutions(authUser);
+    String json = adminResponse.getEntity().toString();
+    assertEquals(200, adminResponse.getStatus());
+    assertNotNull(json);
+  }
+
+  @Test
+  public void testGetInstitutionsForNonAdmin() {
+    List<Institution> institutions = new ArrayList<Institution>();
+    Institution mockInstitution = institutionSetup();
+    institutions.add(mockInstitution);
+    when(userService.findUserByEmail(anyString())).thenReturn(researcherUser);
+    when(institutionService.findAllInstitutions()).thenReturn(institutions);
+    initResource();
+    Response researcherResponse = resource.getInstitutions(authUser);
+    String json = researcherResponse.getEntity().toString();
+    assertEquals(200, researcherResponse.getStatus());
+    assertNotNull(json);
+  }
+
+  @Test
+  public void getInstitutionAdmin() {
+    Institution mockInstitution = institutionSetup();
+    when(userService.findUserByEmail(anyString())).thenReturn(adminUser);
+    when(institutionService.findInstitutionById(anyInt())).thenReturn(mockInstitution);
+    String id = "1";
+    Response adminResponse = resource.getInstitution(authUser, id);
+    String json = adminResponse.getEntity().toString();
+    assertEquals(adminResponse.getStatus(), 200);
+    assertNotNull(json);
+  }
+
+  @Test
+  public void getInstitutionNonAdmin() {
+    Institution mockInstitution = institutionSetup();
+    when(userService.findUserByEmail(anyString())).thenReturn(researcherUser);
+    when(institutionService.findInstitutionById(anyInt())).thenReturn(mockInstitution);
+    String id = "1";
+    Response researcherResponse = resource.getInstitution(authUser, id);
+    String json = researcherResponse.getEntity().toString();
+    assertEquals(200, researcherResponse.getStatus());
+    assertNotNull(json);
```

Comment:
For the non-admin responses, it would be nice to know that the exclusion strategy worked. You can coerce the string back into an object with Gson and check the fields

Reply:
The InstitutionUtil tests covers this since that's where the ```ExclusionStrategy``` and the ```GsonBuilder``` are defined. This way I don't have to repeat this test across all of the ```@PermitAll``` routes. (Though for the non Admin response I just check the json directly since comparing the string is faster to type out than deserializing the ```Institution``` object and then checking each field).